install.psm1

```diff
@@ -0,0 +1,262 @@
+        return $InstallDir
+    }
+
+    throw "Invalid install directory provided '$InstallDir'"
+}
+
+function Get-Temp-Directory() {
+    return $env:temp
+}
+
+function Prepare-Install-Directory([string]$InstallDir) {
+    New-Item -ItemType Directory -Force -Path $InstallDir | Out-Null
+}
+
+function Download-OpenTelemetry([string]$Version, [string]$Path) {
+    $archive = "opentelemetry-dotnet-instrumentation-windows.zip"
+    $dlUrl = "https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/releases/download/$Version/$archive"
+    $dlPath = Join-Path $Path $archive
+
+    Invoke-WebRequest -Uri $dlUrl -OutFile $dlPath
+
+    return $dlPath
+}
+
+function Extract-OpenTelemetry([string]$DlPath, [string] $InstallPath) {
+    Expand-Archive $DlPath $InstallPath -Force
+}
+
+function Setup-OpenTelemetry-Environment([string] $InstallPath) {
+    $target = [System.EnvironmentVariableTarget]::Machine
+
+    # .NET Framework
+    [System.Environment]::SetEnvironmentVariable('COR_PROFILER','{918728DD-259F-4A6A-AC2B-B85E1B658318}', $target)
+    [System.Environment]::SetEnvironmentVariable('COR_PROFILER_PATH_32', (Join-Path $InstallPath "/win-x86/OpenTelemetry.AutoInstrumentation.Native.dll"), $target)
+    [System.Environment]::SetEnvironmentVariable('COR_PROFILER_PATH_64', (Join-Path $InstallPath "/win-x64/OpenTelemetry.AutoInstrumentation.Native.dll"), $target)
+
+    # .NET Core
+    [System.Environment]::SetEnvironmentVariable('CORECLR_PROFILER','{918728DD-259F-4A6A-AC2B-B85E1B658318}', $target)
+    [System.Environment]::SetEnvironmentVariable('CORECLR_PROFILER_PATH_32', (Join-Path $InstallPath "/win-x86/OpenTelemetry.AutoInstrumentation.Native.dll"), $target)
+    [System.Environment]::SetEnvironmentVariable('CORECLR_PROFILER_PATH_64', (Join-Path $InstallPath "/win-x64/OpenTelemetry.AutoInstrumentation.Native.dll"), $target)
+
+    # OpenTelemetry
+    [System.Environment]::SetEnvironmentVariable('OTEL_DOTNET_AUTO_HOME', $InstallPath, $target)
+    [System.Environment]::SetEnvironmentVariable('OTEL_DOTNET_AUTO_INTEGRATIONS_FILE', (Join-Path $InstallPath "integrations.json"), $target)
+}
+
+function Setup-Windows-Service([string]$HomeDir, [string]$WindowsServiceName, [string]$OTelServiceName) {
+    $DOTNET_ADDITIONAL_DEPS = Join-Path $HomeDir "AdditionalDeps"
+    $DOTNET_SHARED_STORE = Join-Path $HomeDir "store"
+    $DOTNET_STARTUP_HOOKS = Join-Path $HomeDir "netcoreapp3.1/OpenTelemetry.AutoInstrumentation.StartupHook.dll"
+    
+    [string[]] $vars = @(
+       "COR_ENABLE_PROFILING=1",
+       "CORECLR_ENABLE_PROFILING=1",
+       # ASP.NET Core
+       "ASPNETCORE_HOSTINGSTARTUPASSEMBLIES=OpenTelemetry.AutoInstrumentation.AspNetCoreBootstrapper",
+       # .NET Common
+       "DOTNET_ADDITIONAL_DEPS=$DOTNET_ADDITIONAL_DEPS",
+       "DOTNET_SHARED_STORE=$DOTNET_SHARED_STORE",
+       "DOTNET_STARTUP_HOOKS=$DOTNET_STARTUP_HOOKS"
+    )
+
+    if(-not [string]::IsNullOrEmpty($OTelServiceName)) {
```

Comment:
This can be simplified to `if (-not $OTelServiceName) {`

Reply:
I think it should be actually `IsNullOrWhiteSpace`, `IsNullOrEmpty` is just not good.

Reply:
https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/pull/1297/commits/60e6e03de1a4f4f4512de94ba9d07acb1846b379